2013-09-24T01:51:28.814Z - error: uncaughtException: notifyAPI is not defined date=Tue Sep 24 2013 09:51:28 GMT+0800 (中国标准时间), pid=1508, uid=null, gid=null, cwd=D:\6\node\fablab, execPath=D:\Program Files\StrongLoop Node\node.exe, version=v0.10.0, argv=[node, D:\6\node\fablab\app], rss=69849088, heapTotal=57179904, heapUsed=33644280, loadavg=[0, 0, 0], uptime=450983.3627995, trace=[column=28, file=D:\6\node\fablab\routes\help_api.js, function=null, line=77, method=null, native=false, column=21, file=D:\6\node\node_modules\async\lib\async.js, function=null, line=548, method=null, native=false, column=13, file=D:\6\node\node_modules\async\lib\async.js, function=null, line=224, method=null, native=false, column=13, file=D:\6\node\node_modules\async\lib\async.js, function=iterate, line=131, method=null, native=false, column=25, file=D:\6\node\node_modules\async\lib\async.js, function=null, line=142, method=null, native=false, column=17, file=D:\6\node\node_modules\async\lib\async.js, function=null, line=226, method=null, native=false, column=34, file=D:\6\node\node_modules\async\lib\async.js, function=null, line=553, method=null, native=false, column=6, file=D:\6\node\fablab\routes\help_api.js, function=null, line=73, method=null, native=false, column=4, file=D:\6\node\fablab\model\course_model.js, function=, line=171, method=null, native=false, column=8, file=D:\6\node\node_modules\mongoose\node_modules\mpromise\lib\promise.js, function=, line=162, method=null, native=false], stack=[ReferenceError: notifyAPI is not defined,     at D:\6\node\fablab\routes\help_api.js:77:28,     at D:\6\node\node_modules\async\lib\async.js:548:21,     at D:\6\node\node_modules\async\lib\async.js:224:13,     at iterate (D:\6\node\node_modules\async\lib\async.js:131:13),     at D:\6\node\node_modules\async\lib\async.js:142:25,     at D:\6\node\node_modules\async\lib\async.js:226:17,     at D:\6\node\node_modules\async\lib\async.js:553:34,     at D:\6\node\fablab\routes\help_api.js:73:6,     at Promise.<anonymous> (D:\6\node\fablab\model\course_model.js:171:4),     at Promise.<anonymous> (D:\6\node\node_modules\mongoose\node_modules\mpromise\lib\promise.js:162:8)]
2013-09-24T02:08:26.771Z - error: uncaughtException: notifyAPI is not defined date=Tue Sep 24 2013 10:08:26 GMT+0800 (中国标准时间), pid=1508, uid=null, gid=null, cwd=D:\6\node\fablab, execPath=D:\Program Files\StrongLoop Node\node.exe, version=v0.10.0, argv=[node, D:\6\node\fablab\app], rss=79073280, heapTotal=63359488, heapUsed=28730392, loadavg=[0, 0, 0], uptime=452001.332024, trace=[column=28, file=D:\6\node\fablab\routes\help_api.js, function=null, line=77, method=null, native=false, column=21, file=D:\6\node\node_modules\async\lib\async.js, function=null, line=548, method=null, native=false, column=13, file=D:\6\node\node_modules\async\lib\async.js, function=null, line=224, method=null, native=false, column=13, file=D:\6\node\node_modules\async\lib\async.js, function=iterate, line=131, method=null, native=false, column=25, file=D:\6\node\node_modules\async\lib\async.js, function=null, line=142, method=null, native=false, column=17, file=D:\6\node\node_modules\async\lib\async.js, function=null, line=226, method=null, native=false, column=34, file=D:\6\node\node_modules\async\lib\async.js, function=null, line=553, method=null, native=false, column=6, file=D:\6\node\fablab\routes\help_api.js, function=null, line=73, method=null, native=false, column=4, file=D:\6\node\fablab\model\course_model.js, function=, line=171, method=null, native=false, column=8, file=D:\6\node\node_modules\mongoose\node_modules\mpromise\lib\promise.js, function=, line=162, method=null, native=false], stack=[ReferenceError: notifyAPI is not defined,     at D:\6\node\fablab\routes\help_api.js:77:28,     at D:\6\node\node_modules\async\lib\async.js:548:21,     at D:\6\node\node_modules\async\lib\async.js:224:13,     at iterate (D:\6\node\node_modules\async\lib\async.js:131:13),     at D:\6\node\node_modules\async\lib\async.js:142:25,     at D:\6\node\node_modules\async\lib\async.js:226:17,     at D:\6\node\node_modules\async\lib\async.js:553:34,     at D:\6\node\fablab\routes\help_api.js:73:6,     at Promise.<anonymous> (D:\6\node\fablab\model\course_model.js:171:4),     at Promise.<anonymous> (D:\6\node\node_modules\mongoose\node_modules\mpromise\lib\promise.js:162:8)]
2013-09-24T06:48:04.790Z - error: uncaughtException: data1 is not defined date=Tue Sep 24 2013 14:48:04 GMT+0800 (中国标准时间), pid=11784, uid=null, gid=null, cwd=D:\6\node\fablab, execPath=D:\Program Files\StrongLoop Node\node.exe, version=v0.10.0, argv=[node, D:\6\node\fablab\app], rss=87453696, heapTotal=67475200, heapUsed=43794816, loadavg=[0, 0, 0], uptime=461365.9726508, trace=[column=38, file=D:\6\node\fablab\routes\course_milestone_api.js, function=null, line=121, method=null, native=false, column=13, file=D:\6\node\node_modules\async\lib\async.js, function=null, line=229, method=null, native=false, column=25, file=D:\6\node\node_modules\async\lib\async.js, function=null, line=116, method=null, native=false, column=16, file=D:\6\node\node_modules\async\lib\async.js, function=null, line=24, method=null, native=false, column=17, file=D:\6\node\node_modules\async\lib\async.js, function=null, line=226, method=null, native=false, column=34, file=D:\6\node\node_modules\async\lib\async.js, function=null, line=513, method=null, native=false, column=17, file=D:\6\node\fablab\routes\course_milestone_api.js, function=null, line=117, method=null, native=false, column=4, file=D:\6\node\fablab\model\student_plan_model.js, function=, line=119, method=null, native=false, column=8, file=D:\6\node\node_modules\mongoose\node_modules\mpromise\lib\promise.js, function=, line=162, method=null, native=false, column=17, file=events.js, function=Promise.EventEmitter.emit, line=95, method=EventEmitter.emit, native=false], stack=[ReferenceError: data1 is not defined,     at D:\6\node\fablab\routes\course_milestone_api.js:121:38,     at D:\6\node\node_modules\async\lib\async.js:229:13,     at D:\6\node\node_modules\async\lib\async.js:116:25,     at D:\6\node\node_modules\async\lib\async.js:24:16,     at D:\6\node\node_modules\async\lib\async.js:226:17,     at D:\6\node\node_modules\async\lib\async.js:513:34,     at D:\6\node\fablab\routes\course_milestone_api.js:117:17,     at Promise.<anonymous> (D:\6\node\fablab\model\student_plan_model.js:119:4),     at Promise.<anonymous> (D:\6\node\node_modules\mongoose\node_modules\mpromise\lib\promise.js:162:8),     at Promise.EventEmitter.emit (events.js:95:17)]
2013-09-24T06:48:10.857Z - error: uncaughtException: data1 is not defined date=Tue Sep 24 2013 14:48:10 GMT+0800 (中国标准时间), pid=11784, uid=null, gid=null, cwd=D:\6\node\fablab, execPath=D:\Program Files\StrongLoop Node\node.exe, version=v0.10.0, argv=[node, D:\6\node\fablab\app], rss=85401600, heapTotal=67475200, heapUsed=39493952, loadavg=[0, 0, 0], uptime=461372.0469982, trace=[column=38, file=D:\6\node\fablab\routes\course_milestone_api.js, function=null, line=121, method=null, native=false, column=13, file=D:\6\node\node_modules\async\lib\async.js, function=null, line=229, method=null, native=false, column=25, file=D:\6\node\node_modules\async\lib\async.js, function=null, line=116, method=null, native=false, column=16, file=D:\6\node\node_modules\async\lib\async.js, function=null, line=24, method=null, native=false, column=17, file=D:\6\node\node_modules\async\lib\async.js, function=null, line=226, method=null, native=false, column=34, file=D:\6\node\node_modules\async\lib\async.js, function=null, line=513, method=null, native=false, column=17, file=D:\6\node\fablab\routes\course_milestone_api.js, function=null, line=117, method=null, native=false, column=4, file=D:\6\node\fablab\model\student_plan_model.js, function=, line=119, method=null, native=false, column=8, file=D:\6\node\node_modules\mongoose\node_modules\mpromise\lib\promise.js, function=, line=162, method=null, native=false, column=17, file=events.js, function=Promise.EventEmitter.emit, line=95, method=EventEmitter.emit, native=false], stack=[ReferenceError: data1 is not defined,     at D:\6\node\fablab\routes\course_milestone_api.js:121:38,     at D:\6\node\node_modules\async\lib\async.js:229:13,     at D:\6\node\node_modules\async\lib\async.js:116:25,     at D:\6\node\node_modules\async\lib\async.js:24:16,     at D:\6\node\node_modules\async\lib\async.js:226:17,     at D:\6\node\node_modules\async\lib\async.js:513:34,     at D:\6\node\fablab\routes\course_milestone_api.js:117:17,     at Promise.<anonymous> (D:\6\node\fablab\model\student_plan_model.js:119:4),     at Promise.<anonymous> (D:\6\node\node_modules\mongoose\node_modules\mpromise\lib\promise.js:162:8),     at Promise.EventEmitter.emit (events.js:95:17)]
2013-09-24T06:49:06.946Z - error: uncaughtException: data1 is not defined date=Tue Sep 24 2013 14:49:06 GMT+0800 (中国标准时间), pid=11784, uid=null, gid=null, cwd=D:\6\node\fablab, execPath=D:\Program Files\StrongLoop Node\node.exe, version=v0.10.0, argv=[node, D:\6\node\fablab\app], rss=85979136, heapTotal=68519168, heapUsed=31024728, loadavg=[0, 0, 0], uptime=461428.1332062, trace=[column=38, file=D:\6\node\fablab\routes\course_milestone_api.js, function=null, line=121, method=null, native=false, column=13, file=D:\6\node\node_modules\async\lib\async.js, function=null, line=229, method=null, native=false, column=25, file=D:\6\node\node_modules\async\lib\async.js, function=null, line=116, method=null, native=false, column=16, file=D:\6\node\node_modules\async\lib\async.js, function=null, line=24, method=null, native=false, column=17, file=D:\6\node\node_modules\async\lib\async.js, function=null, line=226, method=null, native=false, column=34, file=D:\6\node\node_modules\async\lib\async.js, function=null, line=513, method=null, native=false, column=17, file=D:\6\node\fablab\routes\course_milestone_api.js, function=null, line=117, method=null, native=false, column=4, file=D:\6\node\fablab\model\student_plan_model.js, function=, line=119, method=null, native=false, column=8, file=D:\6\node\node_modules\mongoose\node_modules\mpromise\lib\promise.js, function=, line=162, method=null, native=false, column=17, file=events.js, function=Promise.EventEmitter.emit, line=95, method=EventEmitter.emit, native=false], stack=[ReferenceError: data1 is not defined,     at D:\6\node\fablab\routes\course_milestone_api.js:121:38,     at D:\6\node\node_modules\async\lib\async.js:229:13,     at D:\6\node\node_modules\async\lib\async.js:116:25,     at D:\6\node\node_modules\async\lib\async.js:24:16,     at D:\6\node\node_modules\async\lib\async.js:226:17,     at D:\6\node\node_modules\async\lib\async.js:513:34,     at D:\6\node\fablab\routes\course_milestone_api.js:117:17,     at Promise.<anonymous> (D:\6\node\fablab\model\student_plan_model.js:119:4),     at Promise.<anonymous> (D:\6\node\node_modules\mongoose\node_modules\mpromise\lib\promise.js:162:8),     at Promise.EventEmitter.emit (events.js:95:17)]
2013-09-24T12:36:50.075Z - error: uncaughtException: data is not defined date=Tue Sep 24 2013 20:36:50 GMT+0800 (中国标准时间), pid=13272, uid=null, gid=null, cwd=D:\6\node\fablab, execPath=D:\Program Files\StrongLoop Node\node.exe, version=v0.10.0, argv=[node, D:\6\node\fablab\app], rss=76963840, heapTotal=61295616, heapUsed=30693552, loadavg=[0, 0, 0], uptime=480089.6505839, trace=[column=59, file=D:\6\node\fablab\routes\course_milestone_api.js, function=null, line=107, method=null, native=false, column=4, file=D:\6\node\fablab\model\student_plan_model.js, function=, line=119, method=null, native=false, column=8, file=D:\6\node\node_modules\mongoose\node_modules\mpromise\lib\promise.js, function=, line=162, method=null, native=false, column=17, file=events.js, function=Promise.EventEmitter.emit, line=95, method=EventEmitter.emit, native=false, column=38, file=D:\6\node\node_modules\mongoose\node_modules\mpromise\lib\promise.js, function=Promise.emit, line=79, method=emit, native=false, column=20, file=D:\6\node\node_modules\mongoose\node_modules\mpromise\lib\promise.js, function=Promise.fulfill, line=92, method=fulfill, native=false, column=13, file=D:\6\node\node_modules\mongoose\lib\query.js, function=null, line=1823, method=null, native=false, column=11, file=D:\6\node\node_modules\mongoose\lib\document.js, function=model.Document.init, line=251, method=Document.init, native=false, column=10, file=D:\6\node\node_modules\mongoose\lib\query.js, function=completeOne, line=1821, method=null, native=false, column=11, file=D:\6\node\node_modules\mongoose\lib\query.js, function=null, line=1789, method=null, native=false], stack=[ReferenceError: data is not defined,     at D:\6\node\fablab\routes\course_milestone_api.js:107:59,     at Promise.<anonymous> (D:\6\node\fablab\model\student_plan_model.js:119:4),     at Promise.<anonymous> (D:\6\node\node_modules\mongoose\node_modules\mpromise\lib\promise.js:162:8),     at Promise.EventEmitter.emit (events.js:95:17),     at Promise.emit (D:\6\node\node_modules\mongoose\node_modules\mpromise\lib\promise.js:79:38),     at Promise.fulfill (D:\6\node\node_modules\mongoose\node_modules\mpromise\lib\promise.js:92:20),     at D:\6\node\node_modules\mongoose\lib\query.js:1823:13,     at model.Document.init (D:\6\node\node_modules\mongoose\lib\document.js:251:11),     at completeOne (D:\6\node\node_modules\mongoose\lib\query.js:1821:10),     at D:\6\node\node_modules\mongoose\lib\query.js:1789:11]
2013-09-24T12:37:27.090Z - error: uncaughtException: data1 is not defined date=Tue Sep 24 2013 20:37:27 GMT+0800 (中国标准时间), pid=3296, uid=null, gid=null, cwd=D:\6\node\fablab, execPath=D:\Program Files\StrongLoop Node\node.exe, version=v0.10.0, argv=[node, D:\6\node\fablab\app], rss=77455360, heapTotal=60275712, heapUsed=30103696, loadavg=[0, 0, 0], uptime=480126.6617008, trace=[column=26, file=D:\6\node\fablab\routes\course_milestone_api.js, function=null, line=112, method=null, native=false, column=13, file=D:\6\node\node_modules\async\lib\async.js, function=null, line=229, method=null, native=false, column=25, file=D:\6\node\node_modules\async\lib\async.js, function=null, line=116, method=null, native=false, column=16, file=D:\6\node\node_modules\async\lib\async.js, function=null, line=24, method=null, native=false, column=17, file=D:\6\node\node_modules\async\lib\async.js, function=null, line=226, method=null, native=false, column=34, file=D:\6\node\node_modules\async\lib\async.js, function=null, line=513, method=null, native=false, column=5, file=D:\6\node\fablab\routes\course_milestone_api.js, function=null, line=108, method=null, native=false, column=4, file=D:\6\node\fablab\model\student_plan_model.js, function=, line=119, method=null, native=false, column=8, file=D:\6\node\node_modules\mongoose\node_modules\mpromise\lib\promise.js, function=, line=162, method=null, native=false, column=17, file=events.js, function=Promise.EventEmitter.emit, line=95, method=EventEmitter.emit, native=false], stack=[ReferenceError: data1 is not defined,     at D:\6\node\fablab\routes\course_milestone_api.js:112:26,     at D:\6\node\node_modules\async\lib\async.js:229:13,     at D:\6\node\node_modules\async\lib\async.js:116:25,     at D:\6\node\node_modules\async\lib\async.js:24:16,     at D:\6\node\node_modules\async\lib\async.js:226:17,     at D:\6\node\node_modules\async\lib\async.js:513:34,     at D:\6\node\fablab\routes\course_milestone_api.js:108:5,     at Promise.<anonymous> (D:\6\node\fablab\model\student_plan_model.js:119:4),     at Promise.<anonymous> (D:\6\node\node_modules\mongoose\node_modules\mpromise\lib\promise.js:162:8),     at Promise.EventEmitter.emit (events.js:95:17)]
2013-09-24T12:49:09.124Z - error: uncaughtException: Unexpected token o date=Tue Sep 24 2013 20:49:09 GMT+0800 (中国标准时间), pid=1328, uid=null, gid=null, cwd=D:\6\node\fablab, execPath=D:\Program Files\StrongLoop Node\node.exe, version=v0.10.0, argv=[node, D:\6\node\fablab\app], rss=77352960, heapTotal=60275712, heapUsed=30048480, loadavg=[0, 0, 0], uptime=480828.7028553, trace=[column=null, file=null, function=Object.parse, line=null, method=parse, native=true, column=20, file=D:\6\node\fablab\routes\course_milestone_api.js, function=null, line=118, method=null, native=false, column=13, file=D:\6\node\node_modules\async\lib\async.js, function=null, line=229, method=null, native=false, column=25, file=D:\6\node\node_modules\async\lib\async.js, function=null, line=116, method=null, native=false, column=16, file=D:\6\node\node_modules\async\lib\async.js, function=null, line=24, method=null, native=false, column=17, file=D:\6\node\node_modules\async\lib\async.js, function=null, line=226, method=null, native=false, column=34, file=D:\6\node\node_modules\async\lib\async.js, function=null, line=513, method=null, native=false, column=9, file=D:\6\node\fablab\routes\course_milestone_api.js, function=null, line=113, method=null, native=false, column=4, file=D:\6\node\fablab\model\student_plan_model.js, function=, line=119, method=null, native=false, column=8, file=D:\6\node\node_modules\mongoose\node_modules\mpromise\lib\promise.js, function=, line=162, method=null, native=false], stack=[SyntaxError: Unexpected token o,     at Object.parse (native),     at D:\6\node\fablab\routes\course_milestone_api.js:118:20,     at D:\6\node\node_modules\async\lib\async.js:229:13,     at D:\6\node\node_modules\async\lib\async.js:116:25,     at D:\6\node\node_modules\async\lib\async.js:24:16,     at D:\6\node\node_modules\async\lib\async.js:226:17,     at D:\6\node\node_modules\async\lib\async.js:513:34,     at D:\6\node\fablab\routes\course_milestone_api.js:113:9,     at Promise.<anonymous> (D:\6\node\fablab\model\student_plan_model.js:119:4),     at Promise.<anonymous> (D:\6\node\node_modules\mongoose\node_modules\mpromise\lib\promise.js:162:8)]
2013-09-25T09:35:08.043Z - error: uncaughtException: error:28069065:lib(40):UI_set_result:result too small date=Wed Sep 25 2013 17:35:08 GMT+0800 (中国标准时间), pid=7700, uid=null, gid=null, cwd=D:\6\node\fablab, execPath=D:\Program Files\StrongLoop Node\node.exe, version=v0.10.0, argv=[node, D:\6\node\fablab\app], rss=65122304, heapTotal=54108160, heapUsed=31432144, loadavg=[0, 0, 0], uptime=521139.0784769, trace=[column=17, file=crypto.js, function=Object.exports.createCredentials, line=100, method=exports.createCredentials, native=false, column=28, file=tls.js, function=Server, line=1032, method=null, native=false, column=14, file=Server (https.js, function=new, line=35, method=null, native=false, column=10, file=https.js, function=Object.exports.createServer, line=54, method=exports.createServer, native=false, column=33, file=D:\6\node\node_modules\socket.io\lib\socket.io.js, function=Object.exports.listen, line=64, method=exports.listen, native=false, column=18, file=D:\6\node\fablab\routes\websocket_api.js, function=init, line=123, method=null, native=false, column=1, file=D:\6\node\fablab\routes\websocket_api.js, function=, line=118, method=null, native=false, column=26, file=module.js, function=Module._compile, line=456, method=_compile, native=false, column=10, file=module.js, function=Object.Module._extensions..js, line=474, method=Module._extensions..js, native=false, column=32, file=module.js, function=Module.load, line=356, method=load, native=false], stack=[Error: error:28069065:lib(40):UI_set_result:result too small,     at Object.exports.createCredentials (crypto.js:100:17),     at Server (tls.js:1032:28),     at new Server (https.js:35:14),     at Object.exports.createServer (https.js:54:10),     at Object.exports.listen (D:\6\node\node_modules\socket.io\lib\socket.io.js:64:33),     at init (D:\6\node\fablab\routes\websocket_api.js:123:18),     at Object.<anonymous> (D:\6\node\fablab\routes\websocket_api.js:118:1),     at Module._compile (module.js:456:26),     at Object.Module._extensions..js (module.js:474:10),     at Module.load (module.js:356:32)]
2013-09-27T09:26:34.453Z - error: uncaughtException: accept is not defined date=Fri Sep 27 2013 17:26:34 GMT+0800 (中国标准时间), pid=3560, uid=null, gid=null, cwd=D:\6\node\fablab, execPath=D:\Program Files\StrongLoop Node\node.exe, version=v0.10.0, argv=[node, D:\6\node\fablab\app], rss=84180992, heapTotal=66455296, heapUsed=37304296, loadavg=[0, 0, 0], uptime=43547.3621732, trace=[column=19, file=D:\6\node\fablab\routes\websocket_api.js, function=, line=148, method=null, native=false, column=31, file=D:\6\node\node_modules\socket.io\lib\manager.js, function=Manager.authorize, line=910, method=authorize, native=false, column=8, file=D:\6\node\node_modules\socket.io\lib\manager.js, function=Manager.handleHandshake, line=786, method=handleHandshake, native=false, column=12, file=D:\6\node\node_modules\socket.io\lib\manager.js, function=Manager.handleRequest, line=593, method=handleRequest, native=false, column=10, file=D:\6\node\node_modules\socket.io\lib\manager.js, function=, line=119, method=null, native=false, column=17, file=events.js, function=Server.EventEmitter.emit, line=98, method=EventEmitter.emit, native=false, column=12, file=http.js, function=HTTPParser.parser.onIncoming, line=2014, method=parser.onIncoming, native=false, column=23, file=[as onHeadersComplete] (http.js, function=HTTPParser.parserOnHeadersComplete, line=119, method=parserOnHeadersComplete, native=false, column=22, file=http.js, function=CleartextStream.socket.ondata, line=1904, method=socket.ondata, native=false, column=10, file=[as _read] (tls.js, function=CleartextStream.read, line=460, method=read, native=false], stack=[ReferenceError: accept is not defined,     at Manager.<anonymous> (D:\6\node\fablab\routes\websocket_api.js:148:19),     at Manager.authorize (D:\6\node\node_modules\socket.io\lib\manager.js:910:31),     at Manager.handleHandshake (D:\6\node\node_modules\socket.io\lib\manager.js:786:8),     at Manager.handleRequest (D:\6\node\node_modules\socket.io\lib\manager.js:593:12),     at Server.<anonymous> (D:\6\node\node_modules\socket.io\lib\manager.js:119:10),     at Server.EventEmitter.emit (events.js:98:17),     at HTTPParser.parser.onIncoming (http.js:2014:12),     at HTTPParser.parserOnHeadersComplete [as onHeadersComplete] (http.js:119:23),     at CleartextStream.socket.ondata (http.js:1904:22),     at CleartextStream.read [as _read] (tls.js:460:10)]
2013-09-28T10:03:08.317Z - error: uncaughtException: error:06065064:digital envelope routines:EVP_DecryptFinal_ex:bad decrypt date=Sat Sep 28 2013 18:03:08 GMT+0800 (中国标准时间), pid=10996, uid=null, gid=null, cwd=D:\6\node\fablab, execPath=D:\Program Files\StrongLoop Node\node.exe, version=v0.10.0, argv=[node, D:\6\node\fablab\app], rss=73793536, heapTotal=59177720, heapUsed=25323184, loadavg=[0, 0, 0], uptime=100206.7369041, trace=[column=17, file=crypto.js, function=Object.exports.createCredentials, line=100, method=exports.createCredentials, native=false, column=28, file=tls.js, function=Server, line=1032, method=null, native=false, column=14, file=Server (https.js, function=new, line=35, method=null, native=false, column=10, file=https.js, function=Object.exports.createServer, line=54, method=exports.createServer, native=false, column=33, file=D:\6\node\node_modules\socket.io\lib\socket.io.js, function=Object.exports.listen, line=64, method=exports.listen, native=false, column=18, file=D:\6\node\fablab\routes\websocket_api.js, function=init, line=123, method=null, native=false, column=1, file=D:\6\node\fablab\routes\websocket_api.js, function=, line=118, method=null, native=false, column=26, file=module.js, function=Module._compile, line=456, method=_compile, native=false, column=10, file=module.js, function=Object.Module._extensions..js, line=474, method=Module._extensions..js, native=false, column=32, file=module.js, function=Module.load, line=356, method=load, native=false], stack=[Error: error:06065064:digital envelope routines:EVP_DecryptFinal_ex:bad decrypt,     at Object.exports.createCredentials (crypto.js:100:17),     at Server (tls.js:1032:28),     at new Server (https.js:35:14),     at Object.exports.createServer (https.js:54:10),     at Object.exports.listen (D:\6\node\node_modules\socket.io\lib\socket.io.js:64:33),     at init (D:\6\node\fablab\routes\websocket_api.js:123:18),     at Object.<anonymous> (D:\6\node\fablab\routes\websocket_api.js:118:1),     at Module._compile (module.js:456:26),     at Object.Module._extensions..js (module.js:474:10),     at Module.load (module.js:356:32)]
2013-09-28T10:06:26.503Z - error: uncaughtException: error:06065064:digital envelope routines:EVP_DecryptFinal_ex:bad decrypt date=Sat Sep 28 2013 18:06:26 GMT+0800 (中国标准时间), pid=10956, uid=null, gid=null, cwd=D:\6\node\fablab, execPath=D:\Program Files\StrongLoop Node\node.exe, version=v0.10.0, argv=[node, D:\6\node\fablab\app], rss=74358784, heapTotal=59177720, heapUsed=25842736, loadavg=[0, 0, 0], uptime=100404.9232397, trace=[column=17, file=crypto.js, function=Object.exports.createCredentials, line=100, method=exports.createCredentials, native=false, column=28, file=tls.js, function=Server, line=1032, method=null, native=false, column=14, file=Server (https.js, function=new, line=35, method=null, native=false, column=10, file=https.js, function=Object.exports.createServer, line=54, method=exports.createServer, native=false, column=33, file=D:\6\node\node_modules\socket.io\lib\socket.io.js, function=Object.exports.listen, line=64, method=exports.listen, native=false, column=18, file=D:\6\node\fablab\routes\websocket_api.js, function=init, line=123, method=null, native=false, column=1, file=D:\6\node\fablab\routes\websocket_api.js, function=, line=118, method=null, native=false, column=26, file=module.js, function=Module._compile, line=456, method=_compile, native=false, column=10, file=module.js, function=Object.Module._extensions..js, line=474, method=Module._extensions..js, native=false, column=32, file=module.js, function=Module.load, line=356, method=load, native=false], stack=[Error: error:06065064:digital envelope routines:EVP_DecryptFinal_ex:bad decrypt,     at Object.exports.createCredentials (crypto.js:100:17),     at Server (tls.js:1032:28),     at new Server (https.js:35:14),     at Object.exports.createServer (https.js:54:10),     at Object.exports.listen (D:\6\node\node_modules\socket.io\lib\socket.io.js:64:33),     at init (D:\6\node\fablab\routes\websocket_api.js:123:18),     at Object.<anonymous> (D:\6\node\fablab\routes\websocket_api.js:118:1),     at Module._compile (module.js:456:26),     at Object.Module._extensions..js (module.js:474:10),     at Module.load (module.js:356:32)]
2013-09-28T10:06:26.529Z - error: uncaughtException: ENOENT, no such file or directory 'D:\6\node\fablab\certs\server.cert' date=Sat Sep 28 2013 18:06:26 GMT+0800 (中国标准时间), pid=10956, uid=null, gid=null, cwd=D:\6\node\fablab, execPath=D:\Program Files\StrongLoop Node\node.exe, version=v0.10.0, argv=[node, D:\6\node\fablab\app], rss=74596352, heapTotal=60197624, heapUsed=26435648, loadavg=[0, 0, 0], uptime=100404.9552415, trace=[column=18, file=fs.js, function=Object.fs.openSync, line=413, method=fs.openSync, native=false, column=15, file=fs.js, function=Object.fs.readFileSync, line=270, method=fs.readFileSync, native=false, column=30, file=D:\6\node\fablab\app.js, function=loadServer, line=382, method=null, native=false, column=17, file=D:\6\node\fablab\app.js, function=, line=376, method=null, native=false, column=17, file=events.js, function=ChildProcess.EventEmitter.emit, line=98, method=EventEmitter.emit, native=false, column=12, file=child_process.js, function=Process.ChildProcess._handle.onexit, line=754, method=ChildProcess._handle.onexit, native=false], stack=[Error: ENOENT, no such file or directory 'D:\6\node\fablab\certs\server.cert',     at Object.fs.openSync (fs.js:413:18),     at Object.fs.readFileSync (fs.js:270:15),     at loadServer (D:\6\node\fablab\app.js:382:30),     at ChildProcess.<anonymous> (D:\6\node\fablab\app.js:376:17),     at ChildProcess.EventEmitter.emit (events.js:98:17),     at Process.ChildProcess._handle.onexit (child_process.js:754:12)]
2013-09-28T10:06:35.406Z - error: uncaughtException: ENOENT, no such file or directory 'D:\6\node\fablab\certs\server.cert' date=Sat Sep 28 2013 18:06:35 GMT+0800 (中国标准时间), pid=10948, uid=null, gid=null, cwd=D:\6\node\fablab, execPath=D:\Program Files\StrongLoop Node\node.exe, version=v0.10.0, argv=[node, D:\6\node\fablab\app], rss=76058624, heapTotal=60197624, heapUsed=26825728, loadavg=[0, 0, 0], uptime=100413.827749, trace=[column=18, file=fs.js, function=Object.fs.openSync, line=413, method=fs.openSync, native=false, column=15, file=fs.js, function=Object.fs.readFileSync, line=270, method=fs.readFileSync, native=false, column=30, file=D:\6\node\fablab\app.js, function=loadServer, line=381, method=null, native=false, column=17, file=D:\6\node\fablab\app.js, function=, line=375, method=null, native=false, column=17, file=events.js, function=ChildProcess.EventEmitter.emit, line=98, method=EventEmitter.emit, native=false, column=12, file=child_process.js, function=Process.ChildProcess._handle.onexit, line=754, method=ChildProcess._handle.onexit, native=false], stack=[Error: ENOENT, no such file or directory 'D:\6\node\fablab\certs\server.cert',     at Object.fs.openSync (fs.js:413:18),     at Object.fs.readFileSync (fs.js:270:15),     at loadServer (D:\6\node\fablab\app.js:381:30),     at ChildProcess.<anonymous> (D:\6\node\fablab\app.js:375:17),     at ChildProcess.EventEmitter.emit (events.js:98:17),     at Process.ChildProcess._handle.onexit (child_process.js:754:12)]
2013-09-28T10:06:52.128Z - error: uncaughtException: error:28069065:lib(40):UI_set_result:result too small date=Sat Sep 28 2013 18:06:52 GMT+0800 (中国标准时间), pid=11128, uid=null, gid=null, cwd=D:\6\node\fablab, execPath=D:\Program Files\StrongLoop Node\node.exe, version=v0.10.0, argv=[node, D:\6\node\fablab\app], rss=74432512, heapTotal=59177720, heapUsed=25846680, loadavg=[0, 0, 0], uptime=100430.5487054, trace=[column=17, file=crypto.js, function=Object.exports.createCredentials, line=100, method=exports.createCredentials, native=false, column=28, file=tls.js, function=Server, line=1032, method=null, native=false, column=14, file=Server (https.js, function=new, line=35, method=null, native=false, column=10, file=https.js, function=Object.exports.createServer, line=54, method=exports.createServer, native=false, column=33, file=D:\6\node\node_modules\socket.io\lib\socket.io.js, function=Object.exports.listen, line=64, method=exports.listen, native=false, column=18, file=D:\6\node\fablab\routes\websocket_api.js, function=init, line=123, method=null, native=false, column=1, file=D:\6\node\fablab\routes\websocket_api.js, function=, line=118, method=null, native=false, column=26, file=module.js, function=Module._compile, line=456, method=_compile, native=false, column=10, file=module.js, function=Object.Module._extensions..js, line=474, method=Module._extensions..js, native=false, column=32, file=module.js, function=Module.load, line=356, method=load, native=false], stack=[Error: error:28069065:lib(40):UI_set_result:result too small,     at Object.exports.createCredentials (crypto.js:100:17),     at Server (tls.js:1032:28),     at new Server (https.js:35:14),     at Object.exports.createServer (https.js:54:10),     at Object.exports.listen (D:\6\node\node_modules\socket.io\lib\socket.io.js:64:33),     at init (D:\6\node\fablab\routes\websocket_api.js:123:18),     at Object.<anonymous> (D:\6\node\fablab\routes\websocket_api.js:118:1),     at Module._compile (module.js:456:26),     at Object.Module._extensions..js (module.js:474:10),     at Module.load (module.js:356:32)]
2013-09-28T10:06:52.158Z - error: uncaughtException: ENOENT, no such file or directory 'D:\6\node\fablab\certs\server.cert' date=Sat Sep 28 2013 18:06:52 GMT+0800 (中国标准时间), pid=11128, uid=null, gid=null, cwd=D:\6\node\fablab, execPath=D:\Program Files\StrongLoop Node\node.exe, version=v0.10.0, argv=[node, D:\6\node\fablab\app], rss=74682368, heapTotal=60197624, heapUsed=26433640, loadavg=[0, 0, 0], uptime=100430.5837074, trace=[column=18, file=fs.js, function=Object.fs.openSync, line=413, method=fs.openSync, native=false, column=15, file=fs.js, function=Object.fs.readFileSync, line=270, method=fs.readFileSync, native=false, column=30, file=D:\6\node\fablab\app.js, function=loadServer, line=381, method=null, native=false, column=17, file=D:\6\node\fablab\app.js, function=, line=375, method=null, native=false, column=17, file=events.js, function=ChildProcess.EventEmitter.emit, line=98, method=EventEmitter.emit, native=false, column=12, file=child_process.js, function=Process.ChildProcess._handle.onexit, line=754, method=ChildProcess._handle.onexit, native=false], stack=[Error: ENOENT, no such file or directory 'D:\6\node\fablab\certs\server.cert',     at Object.fs.openSync (fs.js:413:18),     at Object.fs.readFileSync (fs.js:270:15),     at loadServer (D:\6\node\fablab\app.js:381:30),     at ChildProcess.<anonymous> (D:\6\node\fablab\app.js:375:17),     at ChildProcess.EventEmitter.emit (events.js:98:17),     at Process.ChildProcess._handle.onexit (child_process.js:754:12)]
2013-09-28T10:08:10.008Z - error: uncaughtException: error:28069065:lib(40):UI_set_result:result too small date=Sat Sep 28 2013 18:08:09 GMT+0800 (中国标准时间), pid=7696, uid=null, gid=null, cwd=D:\6\node\fablab, execPath=D:\Program Files\StrongLoop Node\node.exe, version=v0.10.0, argv=[node, D:\6\node\fablab\app], rss=74412032, heapTotal=59177720, heapUsed=25739680, loadavg=[0, 0, 0], uptime=100508.4281598, trace=[column=17, file=crypto.js, function=Object.exports.createCredentials, line=100, method=exports.createCredentials, native=false, column=28, file=tls.js, function=Server, line=1032, method=null, native=false, column=14, file=Server (https.js, function=new, line=35, method=null, native=false, column=10, file=https.js, function=Object.exports.createServer, line=54, method=exports.createServer, native=false, column=33, file=D:\6\node\node_modules\socket.io\lib\socket.io.js, function=Object.exports.listen, line=64, method=exports.listen, native=false, column=18, file=D:\6\node\fablab\routes\websocket_api.js, function=init, line=123, method=null, native=false, column=1, file=D:\6\node\fablab\routes\websocket_api.js, function=, line=118, method=null, native=false, column=26, file=module.js, function=Module._compile, line=456, method=_compile, native=false, column=10, file=module.js, function=Object.Module._extensions..js, line=474, method=Module._extensions..js, native=false, column=32, file=module.js, function=Module.load, line=356, method=load, native=false], stack=[Error: error:28069065:lib(40):UI_set_result:result too small,     at Object.exports.createCredentials (crypto.js:100:17),     at Server (tls.js:1032:28),     at new Server (https.js:35:14),     at Object.exports.createServer (https.js:54:10),     at Object.exports.listen (D:\6\node\node_modules\socket.io\lib\socket.io.js:64:33),     at init (D:\6\node\fablab\routes\websocket_api.js:123:18),     at Object.<anonymous> (D:\6\node\fablab\routes\websocket_api.js:118:1),     at Module._compile (module.js:456:26),     at Object.Module._extensions..js (module.js:474:10),     at Module.load (module.js:356:32)]
2013-09-28T10:11:04.394Z - error: uncaughtException: error:28069065:lib(40):UI_set_result:result too small date=Sat Sep 28 2013 18:11:04 GMT+0800 (中国标准时间), pid=9988, uid=null, gid=null, cwd=D:\6\node\fablab, execPath=D:\Program Files\StrongLoop Node\node.exe, version=v0.10.0, argv=[node, D:\6\node\fablab\app], rss=74375168, heapTotal=59177720, heapUsed=25739792, loadavg=[0, 0, 0], uptime=100682.8131341, trace=[column=17, file=crypto.js, function=Object.exports.createCredentials, line=100, method=exports.createCredentials, native=false, column=28, file=tls.js, function=Server, line=1032, method=null, native=false, column=14, file=Server (https.js, function=new, line=35, method=null, native=false, column=10, file=https.js, function=Object.exports.createServer, line=54, method=exports.createServer, native=false, column=33, file=D:\6\node\node_modules\socket.io\lib\socket.io.js, function=Object.exports.listen, line=64, method=exports.listen, native=false, column=18, file=D:\6\node\fablab\routes\websocket_api.js, function=init, line=123, method=null, native=false, column=1, file=D:\6\node\fablab\routes\websocket_api.js, function=, line=118, method=null, native=false, column=26, file=module.js, function=Module._compile, line=456, method=_compile, native=false, column=10, file=module.js, function=Object.Module._extensions..js, line=474, method=Module._extensions..js, native=false, column=32, file=module.js, function=Module.load, line=356, method=load, native=false], stack=[Error: error:28069065:lib(40):UI_set_result:result too small,     at Object.exports.createCredentials (crypto.js:100:17),     at Server (tls.js:1032:28),     at new Server (https.js:35:14),     at Object.exports.createServer (https.js:54:10),     at Object.exports.listen (D:\6\node\node_modules\socket.io\lib\socket.io.js:64:33),     at init (D:\6\node\fablab\routes\websocket_api.js:123:18),     at Object.<anonymous> (D:\6\node\fablab\routes\websocket_api.js:118:1),     at Module._compile (module.js:456:26),     at Object.Module._extensions..js (module.js:474:10),     at Module.load (module.js:356:32)]
2013-09-28T10:14:54.790Z - error: uncaughtException: ENOENT, no such file or directory 'D:\6\node\fablab\certs\server.cert' date=Sat Sep 28 2013 18:14:54 GMT+0800 (中国标准时间), pid=2104, uid=null, gid=null, cwd=D:\6\node\fablab, execPath=D:\Program Files\StrongLoop Node\node.exe, version=v0.10.0, argv=[node, D:\6\node\fablab\app], rss=76005376, heapTotal=60197624, heapUsed=26425088, loadavg=[0, 0, 0], uptime=100913.209312, trace=[column=18, file=fs.js, function=Object.fs.openSync, line=413, method=fs.openSync, native=false, column=15, file=fs.js, function=Object.fs.readFileSync, line=270, method=fs.readFileSync, native=false, column=30, file=D:\6\node\fablab\app.js, function=loadServer, line=382, method=null, native=false, column=17, file=D:\6\node\fablab\app.js, function=, line=376, method=null, native=false, column=17, file=events.js, function=ChildProcess.EventEmitter.emit, line=98, method=EventEmitter.emit, native=false, column=12, file=child_process.js, function=Process.ChildProcess._handle.onexit, line=754, method=ChildProcess._handle.onexit, native=false], stack=[Error: ENOENT, no such file or directory 'D:\6\node\fablab\certs\server.cert',     at Object.fs.openSync (fs.js:413:18),     at Object.fs.readFileSync (fs.js:270:15),     at loadServer (D:\6\node\fablab\app.js:382:30),     at ChildProcess.<anonymous> (D:\6\node\fablab\app.js:376:17),     at ChildProcess.EventEmitter.emit (events.js:98:17),     at Process.ChildProcess._handle.onexit (child_process.js:754:12)]
2013-09-28T22:13:04.339Z - error: uncaughtException: Offset is out of bounds date=Sun Sep 29 2013 06:13:04 GMT+0800 (中国标准时间), pid=11156, uid=null, gid=null, cwd=D:\6\node\fablab, execPath=D:\Program Files\StrongLoop Node\node.exe, version=v0.10.0, argv=[node, D:\6\node\fablab\app], rss=102998016, heapTotal=77704440, heapUsed=45910768, loadavg=[0, 0, 0], uptime=138464.2311032, trace=[column=11, file=fs.js, function=Object.fs.read, line=443, method=fs.read, native=false, column=12, file=[as _onImmediate] (D:\6\node\fablab\node_modules\mongoose\node_modules\mongodb\lib\mongodb\gridfs\gridstore.js, function=Object.writeChunk, line=325, method=writeChunk, native=false, column=15, file=[as _immediateCallback] (timers.js, function=processImmediate, line=309, method=null, native=false], stack=[Error: Offset is out of bounds,     at Object.fs.read (fs.js:443:11),     at Object.writeChunk [as _onImmediate] (D:\6\node\fablab\node_modules\mongoose\node_modules\mongodb\lib\mongodb\gridfs\gridstore.js:325:12),     at processImmediate [as _immediateCallback] (timers.js:309:15)]
2013-09-29T10:56:47.074Z - error: uncaughtException: next is not defined date=Sun Sep 29 2013 18:56:47 GMT+0800 (中国标准时间), pid=18056, uid=null, gid=null, cwd=D:\6\node\fablab, execPath=D:\Program Files\StrongLoop Node\node.exe, version=v0.10.0, argv=[node, D:\6\node\fablab\app], rss=92180480, heapTotal=74608632, heapUsed=34956960, loadavg=[0, 0, 0], uptime=176486.9708795, trace=[column=17, file=D:\6\node\fablab\routes\course_milestone_api.js, function=null, line=99, method=null, native=false, column=4, file=D:\6\node\fablab\model\student_plan_model.js, function=null, line=46, method=null, native=false, column=16, file=D:\6\node\fablab\node_modules\mongoose\lib\utils.js, function=null, line=414, method=null, native=false, column=9, file=D:\6\node\fablab\node_modules\mongoose\node_modules\mongodb\lib\mongodb\collection.js, function=null, line=494, method=null, native=false, column=41, file=D:\6\node\fablab\node_modules\mongoose\node_modules\mongodb\lib\mongodb\connection\base.js, function=Server.Base._callHandler, line=382, method=Base._callHandler, native=false, column=18, file=D:\6\node\fablab\node_modules\mongoose\node_modules\mongodb\lib\mongodb\connection\server.js, function=null, line=472, method=null, native=false, column=5, file=D:\6\node\fablab\node_modules\mongoose\node_modules\mongodb\lib\mongodb\responses\mongo_reply.js, function=MongoReply.parseBody, line=68, method=parseBody, native=false, column=20, file=D:\6\node\fablab\node_modules\mongoose\node_modules\mongodb\lib\mongodb\connection\server.js, function=, line=430, method=null, native=false, column=17, file=events.js, function=EventEmitter.emit, line=95, method=emit, native=false, column=13, file=D:\6\node\fablab\node_modules\mongoose\node_modules\mongodb\lib\mongodb\connection\connection_pool.js, function=, line=191, method=null, native=false], stack=[ReferenceError: next is not defined,     at D:\6\node\fablab\routes\course_milestone_api.js:99:17,     at D:\6\node\fablab\model\student_plan_model.js:46:4,     at D:\6\node\fablab\node_modules\mongoose\lib\utils.js:414:16,     at D:\6\node\fablab\node_modules\mongoose\node_modules\mongodb\lib\mongodb\collection.js:494:9,     at Server.Base._callHandler (D:\6\node\fablab\node_modules\mongoose\node_modules\mongodb\lib\mongodb\connection\base.js:382:41),     at D:\6\node\fablab\node_modules\mongoose\node_modules\mongodb\lib\mongodb\connection\server.js:472:18,     at MongoReply.parseBody (D:\6\node\fablab\node_modules\mongoose\node_modules\mongodb\lib\mongodb\responses\mongo_reply.js:68:5),     at null.<anonymous> (D:\6\node\fablab\node_modules\mongoose\node_modules\mongodb\lib\mongodb\connection\server.js:430:20),     at EventEmitter.emit (events.js:95:17),     at null.<anonymous> (D:\6\node\fablab\node_modules\mongoose\node_modules\mongodb\lib\mongodb\connection\connection_pool.js:191:13)]
2013-09-30T00:37:47.744Z - error: uncaughtException: accept is not defined date=Mon Sep 30 2013 08:37:47 GMT+0800 (中国标准时间), pid=6244, uid=null, gid=null, cwd=D:\6\node\fablab, execPath=D:\Program Files\StrongLoop Node\node.exe, version=v0.10.0, argv=[node, D:\6\node\fablab\app], rss=114008064, heapTotal=86991864, heapUsed=56042896, loadavg=[0, 0, 0], uptime=210857.2117422, trace=[column=19, file=D:\6\node\fablab\routes\websocket_api.js, function=, line=148, method=null, native=false, column=31, file=D:\6\node\node_modules\socket.io\lib\manager.js, function=Manager.authorize, line=910, method=authorize, native=false, column=8, file=D:\6\node\node_modules\socket.io\lib\manager.js, function=Manager.handleHandshake, line=786, method=handleHandshake, native=false, column=12, file=D:\6\node\node_modules\socket.io\lib\manager.js, function=Manager.handleRequest, line=593, method=handleRequest, native=false, column=10, file=D:\6\node\node_modules\socket.io\lib\manager.js, function=, line=119, method=null, native=false, column=17, file=events.js, function=Server.EventEmitter.emit, line=98, method=EventEmitter.emit, native=false, column=12, file=http.js, function=HTTPParser.parser.onIncoming, line=2014, method=parser.onIncoming, native=false, column=23, file=[as onHeadersComplete] (http.js, function=HTTPParser.parserOnHeadersComplete, line=119, method=parserOnHeadersComplete, native=false, column=22, file=http.js, function=CleartextStream.socket.ondata, line=1904, method=socket.ondata, native=false, column=10, file=[as _read] (tls.js, function=CleartextStream.read, line=460, method=read, native=false], stack=[ReferenceError: accept is not defined,     at Manager.<anonymous> (D:\6\node\fablab\routes\websocket_api.js:148:19),     at Manager.authorize (D:\6\node\node_modules\socket.io\lib\manager.js:910:31),     at Manager.handleHandshake (D:\6\node\node_modules\socket.io\lib\manager.js:786:8),     at Manager.handleRequest (D:\6\node\node_modules\socket.io\lib\manager.js:593:12),     at Server.<anonymous> (D:\6\node\node_modules\socket.io\lib\manager.js:119:10),     at Server.EventEmitter.emit (events.js:98:17),     at HTTPParser.parser.onIncoming (http.js:2014:12),     at HTTPParser.parserOnHeadersComplete [as onHeadersComplete] (http.js:119:23),     at CleartextStream.socket.ondata (http.js:1904:22),     at CleartextStream.read [as _read] (tls.js:460:10)]
2013-09-30T14:26:58.943Z - error: uncaughtException: Cannot read property 'plan' of null date=Mon Sep 30 2013 22:26:58 GMT+0800 (中国标准时间), pid=4676, uid=null, gid=null, cwd=D:\6\node\fablab, execPath=D:\Program Files\StrongLoop Node\node.exe, version=v0.10.0, argv=[node, D:\6\node\fablab\app], rss=112549888, heapTotal=86979832, heapUsed=49509040, loadavg=[0, 0, 0], uptime=239629.2774103, trace=[column=20, file=D:\6\node\fablab\routes\course_milestone_api.js, function=null, line=272, method=null, native=false, column=4, file=D:\6\node\fablab\model\student_plan_model.js, function=, line=138, method=null, native=false, column=8, file=D:\6\node\fablab\node_modules\mongoose\node_modules\mpromise\lib\promise.js, function=, line=162, method=null, native=false, column=17, file=events.js, function=Promise.EventEmitter.emit, line=95, method=EventEmitter.emit, native=false, column=38, file=D:\6\node\fablab\node_modules\mongoose\node_modules\mpromise\lib\promise.js, function=Promise.emit, line=79, method=emit, native=false, column=20, file=D:\6\node\fablab\node_modules\mongoose\node_modules\mpromise\lib\promise.js, function=Promise.fulfill, line=92, method=fulfill, native=false, column=30, file=D:\6\node\fablab\node_modules\mongoose\lib\query.js, function=null, line=1794, method=null, native=false, column=16, file=D:\6\node\fablab\node_modules\mongoose\lib\utils.js, function=null, line=414, method=null, native=false, column=5, file=D:\6\node\fablab\node_modules\mongoose\node_modules\mongodb\lib\mongodb\collection.js, function=null, line=953, method=null, native=false, column=35, file=D:\6\node\fablab\node_modules\mongoose\node_modules\mongodb\lib\mongodb\cursor.js, function=null, line=683, method=null, native=false], stack=[TypeError: Cannot read property 'plan' of null,     at D:\6\node\fablab\routes\course_milestone_api.js:272:20,     at Promise.<anonymous> (D:\6\node\fablab\model\student_plan_model.js:138:4),     at Promise.<anonymous> (D:\6\node\fablab\node_modules\mongoose\node_modules\mpromise\lib\promise.js:162:8),     at Promise.EventEmitter.emit (events.js:95:17),     at Promise.emit (D:\6\node\fablab\node_modules\mongoose\node_modules\mpromise\lib\promise.js:79:38),     at Promise.fulfill (D:\6\node\fablab\node_modules\mongoose\node_modules\mpromise\lib\promise.js:92:20),     at D:\6\node\fablab\node_modules\mongoose\lib\query.js:1794:30,     at D:\6\node\fablab\node_modules\mongoose\lib\utils.js:414:16,     at D:\6\node\fablab\node_modules\mongoose\node_modules\mongodb\lib\mongodb\collection.js:953:5,     at D:\6\node\fablab\node_modules\mongoose\node_modules\mongodb\lib\mongodb\cursor.js:683:35]
2013-09-30T14:27:42.669Z - error: uncaughtException: Cannot read property 'plan' of null date=Mon Sep 30 2013 22:27:42 GMT+0800 (中国标准时间), pid=4676, uid=null, gid=null, cwd=D:\6\node\fablab, execPath=D:\Program Files\StrongLoop Node\node.exe, version=v0.10.0, argv=[node, D:\6\node\fablab\app], rss=113029120, heapTotal=86979832, heapUsed=53966136, loadavg=[0, 0, 0], uptime=239673.0089116, trace=[column=20, file=D:\6\node\fablab\routes\course_milestone_api.js, function=null, line=272, method=null, native=false, column=4, file=D:\6\node\fablab\model\student_plan_model.js, function=, line=138, method=null, native=false, column=8, file=D:\6\node\fablab\node_modules\mongoose\node_modules\mpromise\lib\promise.js, function=, line=162, method=null, native=false, column=17, file=events.js, function=Promise.EventEmitter.emit, line=95, method=EventEmitter.emit, native=false, column=38, file=D:\6\node\fablab\node_modules\mongoose\node_modules\mpromise\lib\promise.js, function=Promise.emit, line=79, method=emit, native=false, column=20, file=D:\6\node\fablab\node_modules\mongoose\node_modules\mpromise\lib\promise.js, function=Promise.fulfill, line=92, method=fulfill, native=false, column=30, file=D:\6\node\fablab\node_modules\mongoose\lib\query.js, function=null, line=1794, method=null, native=false, column=16, file=D:\6\node\fablab\node_modules\mongoose\lib\utils.js, function=null, line=414, method=null, native=false, column=5, file=D:\6\node\fablab\node_modules\mongoose\node_modules\mongodb\lib\mongodb\collection.js, function=null, line=953, method=null, native=false, column=35, file=D:\6\node\fablab\node_modules\mongoose\node_modules\mongodb\lib\mongodb\cursor.js, function=null, line=683, method=null, native=false], stack=[TypeError: Cannot read property 'plan' of null,     at D:\6\node\fablab\routes\course_milestone_api.js:272:20,     at Promise.<anonymous> (D:\6\node\fablab\model\student_plan_model.js:138:4),     at Promise.<anonymous> (D:\6\node\fablab\node_modules\mongoose\node_modules\mpromise\lib\promise.js:162:8),     at Promise.EventEmitter.emit (events.js:95:17),     at Promise.emit (D:\6\node\fablab\node_modules\mongoose\node_modules\mpromise\lib\promise.js:79:38),     at Promise.fulfill (D:\6\node\fablab\node_modules\mongoose\node_modules\mpromise\lib\promise.js:92:20),     at D:\6\node\fablab\node_modules\mongoose\lib\query.js:1794:30,     at D:\6\node\fablab\node_modules\mongoose\lib\utils.js:414:16,     at D:\6\node\fablab\node_modules\mongoose\node_modules\mongodb\lib\mongodb\collection.js:953:5,     at D:\6\node\fablab\node_modules\mongoose\node_modules\mongodb\lib\mongodb\cursor.js:683:35]
2013-09-30T14:37:21.635Z - error: uncaughtException: Cannot read property 'plan' of null date=Mon Sep 30 2013 22:37:21 GMT+0800 (中国标准时间), pid=6428, uid=null, gid=null, cwd=D:\6\node\fablab, execPath=D:\Program Files\StrongLoop Node\node.exe, version=v0.10.0, argv=[node, D:\6\node\fablab\app], rss=84484096, heapTotal=66365176, heapUsed=41362680, loadavg=[0, 0, 0], uptime=240251.9630258, trace=[column=20, file=D:\6\node\fablab\routes\course_milestone_api.js, function=null, line=272, method=null, native=false, column=4, file=D:\6\node\fablab\model\student_plan_model.js, function=, line=138, method=null, native=false, column=8, file=D:\6\node\fablab\node_modules\mongoose\node_modules\mpromise\lib\promise.js, function=, line=162, method=null, native=false, column=17, file=events.js, function=Promise.EventEmitter.emit, line=95, method=EventEmitter.emit, native=false, column=38, file=D:\6\node\fablab\node_modules\mongoose\node_modules\mpromise\lib\promise.js, function=Promise.emit, line=79, method=emit, native=false, column=20, file=D:\6\node\fablab\node_modules\mongoose\node_modules\mpromise\lib\promise.js, function=Promise.fulfill, line=92, method=fulfill, native=false, column=30, file=D:\6\node\fablab\node_modules\mongoose\lib\query.js, function=null, line=1794, method=null, native=false, column=16, file=D:\6\node\fablab\node_modules\mongoose\lib\utils.js, function=null, line=414, method=null, native=false, column=5, file=D:\6\node\fablab\node_modules\mongoose\node_modules\mongodb\lib\mongodb\collection.js, function=null, line=953, method=null, native=false, column=35, file=D:\6\node\fablab\node_modules\mongoose\node_modules\mongodb\lib\mongodb\cursor.js, function=null, line=683, method=null, native=false], stack=[TypeError: Cannot read property 'plan' of null,     at D:\6\node\fablab\routes\course_milestone_api.js:272:20,     at Promise.<anonymous> (D:\6\node\fablab\model\student_plan_model.js:138:4),     at Promise.<anonymous> (D:\6\node\fablab\node_modules\mongoose\node_modules\mpromise\lib\promise.js:162:8),     at Promise.EventEmitter.emit (events.js:95:17),     at Promise.emit (D:\6\node\fablab\node_modules\mongoose\node_modules\mpromise\lib\promise.js:79:38),     at Promise.fulfill (D:\6\node\fablab\node_modules\mongoose\node_modules\mpromise\lib\promise.js:92:20),     at D:\6\node\fablab\node_modules\mongoose\lib\query.js:1794:30,     at D:\6\node\fablab\node_modules\mongoose\lib\utils.js:414:16,     at D:\6\node\fablab\node_modules\mongoose\node_modules\mongodb\lib\mongodb\collection.js:953:5,     at D:\6\node\fablab\node_modules\mongoose\node_modules\mongodb\lib\mongodb\cursor.js:683:35]
2013-09-30T14:37:27.166Z - error: uncaughtException: data is not defined date=Mon Sep 30 2013 22:37:27 GMT+0800 (中国标准时间), pid=6428, uid=null, gid=null, cwd=D:\6\node\fablab, execPath=D:\Program Files\StrongLoop Node\node.exe, version=v0.10.0, argv=[node, D:\6\node\fablab\app], rss=85676032, heapTotal=68417016, heapUsed=31640128, loadavg=[0, 0, 0], uptime=240257.4993425, trace=[column=17, file=D:\6\node\fablab\routes\course_student_api.js, function=null, line=184, method=null, native=false, column=4, file=D:\6\node\fablab\model\student_plan_model.js, function=null, line=50, method=null, native=false, column=16, file=D:\6\node\fablab\node_modules\mongoose\lib\utils.js, function=null, line=414, method=null, native=false, column=9, file=D:\6\node\fablab\node_modules\mongoose\node_modules\mongodb\lib\mongodb\collection.js, function=null, line=497, method=null, native=false, column=41, file=D:\6\node\fablab\node_modules\mongoose\node_modules\mongodb\lib\mongodb\connection\base.js, function=Server.Base._callHandler, line=382, method=Base._callHandler, native=false, column=18, file=D:\6\node\fablab\node_modules\mongoose\node_modules\mongodb\lib\mongodb\connection\server.js, function=null, line=472, method=null, native=false, column=5, file=D:\6\node\fablab\node_modules\mongoose\node_modules\mongodb\lib\mongodb\responses\mongo_reply.js, function=MongoReply.parseBody, line=68, method=parseBody, native=false, column=20, file=D:\6\node\fablab\node_modules\mongoose\node_modules\mongodb\lib\mongodb\connection\server.js, function=, line=430, method=null, native=false, column=17, file=events.js, function=EventEmitter.emit, line=95, method=emit, native=false, column=13, file=D:\6\node\fablab\node_modules\mongoose\node_modules\mongodb\lib\mongodb\connection\connection_pool.js, function=, line=191, method=null, native=false], stack=[ReferenceError: data is not defined,     at D:\6\node\fablab\routes\course_student_api.js:184:17,     at D:\6\node\fablab\model\student_plan_model.js:50:4,     at D:\6\node\fablab\node_modules\mongoose\lib\utils.js:414:16,     at D:\6\node\fablab\node_modules\mongoose\node_modules\mongodb\lib\mongodb\collection.js:497:9,     at Server.Base._callHandler (D:\6\node\fablab\node_modules\mongoose\node_modules\mongodb\lib\mongodb\connection\base.js:382:41),     at D:\6\node\fablab\node_modules\mongoose\node_modules\mongodb\lib\mongodb\connection\server.js:472:18,     at MongoReply.parseBody (D:\6\node\fablab\node_modules\mongoose\node_modules\mongodb\lib\mongodb\responses\mongo_reply.js:68:5),     at null.<anonymous> (D:\6\node\fablab\node_modules\mongoose\node_modules\mongodb\lib\mongodb\connection\server.js:430:20),     at EventEmitter.emit (events.js:95:17),     at null.<anonymous> (D:\6\node\fablab\node_modules\mongoose\node_modules\mongodb\lib\mongodb\connection\connection_pool.js:191:13)]
2013-09-30T14:37:49.981Z - error: uncaughtException: data is not defined date=Mon Sep 30 2013 22:37:49 GMT+0800 (中国标准时间), pid=6428, uid=null, gid=null, cwd=D:\6\node\fablab, execPath=D:\Program Files\StrongLoop Node\node.exe, version=v0.10.0, argv=[node, D:\6\node\fablab\app], rss=87814144, heapTotal=69448952, heapUsed=41953200, loadavg=[0, 0, 0], uptime=240280.3136474, trace=[column=17, file=D:\6\node\fablab\routes\course_student_api.js, function=null, line=184, method=null, native=false, column=4, file=D:\6\node\fablab\model\student_plan_model.js, function=null, line=50, method=null, native=false, column=16, file=D:\6\node\fablab\node_modules\mongoose\lib\utils.js, function=null, line=414, method=null, native=false, column=9, file=D:\6\node\fablab\node_modules\mongoose\node_modules\mongodb\lib\mongodb\collection.js, function=null, line=497, method=null, native=false, column=41, file=D:\6\node\fablab\node_modules\mongoose\node_modules\mongodb\lib\mongodb\connection\base.js, function=Server.Base._callHandler, line=382, method=Base._callHandler, native=false, column=18, file=D:\6\node\fablab\node_modules\mongoose\node_modules\mongodb\lib\mongodb\connection\server.js, function=null, line=472, method=null, native=false, column=5, file=D:\6\node\fablab\node_modules\mongoose\node_modules\mongodb\lib\mongodb\responses\mongo_reply.js, function=MongoReply.parseBody, line=68, method=parseBody, native=false, column=20, file=D:\6\node\fablab\node_modules\mongoose\node_modules\mongodb\lib\mongodb\connection\server.js, function=, line=430, method=null, native=false, column=17, file=events.js, function=EventEmitter.emit, line=95, method=emit, native=false, column=13, file=D:\6\node\fablab\node_modules\mongoose\node_modules\mongodb\lib\mongodb\connection\connection_pool.js, function=, line=191, method=null, native=false], stack=[ReferenceError: data is not defined,     at D:\6\node\fablab\routes\course_student_api.js:184:17,     at D:\6\node\fablab\model\student_plan_model.js:50:4,     at D:\6\node\fablab\node_modules\mongoose\lib\utils.js:414:16,     at D:\6\node\fablab\node_modules\mongoose\node_modules\mongodb\lib\mongodb\collection.js:497:9,     at Server.Base._callHandler (D:\6\node\fablab\node_modules\mongoose\node_modules\mongodb\lib\mongodb\connection\base.js:382:41),     at D:\6\node\fablab\node_modules\mongoose\node_modules\mongodb\lib\mongodb\connection\server.js:472:18,     at MongoReply.parseBody (D:\6\node\fablab\node_modules\mongoose\node_modules\mongodb\lib\mongodb\responses\mongo_reply.js:68:5),     at null.<anonymous> (D:\6\node\fablab\node_modules\mongoose\node_modules\mongodb\lib\mongodb\connection\server.js:430:20),     at EventEmitter.emit (events.js:95:17),     at null.<anonymous> (D:\6\node\fablab\node_modules\mongoose\node_modules\mongodb\lib\mongodb\connection\connection_pool.js:191:13)]
2013-09-30T14:38:01.890Z - error: uncaughtException: data is not defined date=Mon Sep 30 2013 22:38:01 GMT+0800 (中国标准时间), pid=6428, uid=null, gid=null, cwd=D:\6\node\fablab, execPath=D:\Program Files\StrongLoop Node\node.exe, version=v0.10.0, argv=[node, D:\6\node\fablab\app], rss=89989120, heapTotal=71512824, heapUsed=43712104, loadavg=[0, 0, 0], uptime=240292.2253287, trace=[column=17, file=D:\6\node\fablab\routes\course_student_api.js, function=null, line=184, method=null, native=false, column=4, file=D:\6\node\fablab\model\student_plan_model.js, function=null, line=50, method=null, native=false, column=16, file=D:\6\node\fablab\node_modules\mongoose\lib\utils.js, function=null, line=414, method=null, native=false, column=9, file=D:\6\node\fablab\node_modules\mongoose\node_modules\mongodb\lib\mongodb\collection.js, function=null, line=497, method=null, native=false, column=41, file=D:\6\node\fablab\node_modules\mongoose\node_modules\mongodb\lib\mongodb\connection\base.js, function=Server.Base._callHandler, line=382, method=Base._callHandler, native=false, column=18, file=D:\6\node\fablab\node_modules\mongoose\node_modules\mongodb\lib\mongodb\connection\server.js, function=null, line=472, method=null, native=false, column=5, file=D:\6\node\fablab\node_modules\mongoose\node_modules\mongodb\lib\mongodb\responses\mongo_reply.js, function=MongoReply.parseBody, line=68, method=parseBody, native=false, column=20, file=D:\6\node\fablab\node_modules\mongoose\node_modules\mongodb\lib\mongodb\connection\server.js, function=, line=430, method=null, native=false, column=17, file=events.js, function=EventEmitter.emit, line=95, method=emit, native=false, column=13, file=D:\6\node\fablab\node_modules\mongoose\node_modules\mongodb\lib\mongodb\connection\connection_pool.js, function=, line=191, method=null, native=false], stack=[ReferenceError: data is not defined,     at D:\6\node\fablab\routes\course_student_api.js:184:17,     at D:\6\node\fablab\model\student_plan_model.js:50:4,     at D:\6\node\fablab\node_modules\mongoose\lib\utils.js:414:16,     at D:\6\node\fablab\node_modules\mongoose\node_modules\mongodb\lib\mongodb\collection.js:497:9,     at Server.Base._callHandler (D:\6\node\fablab\node_modules\mongoose\node_modules\mongodb\lib\mongodb\connection\base.js:382:41),     at D:\6\node\fablab\node_modules\mongoose\node_modules\mongodb\lib\mongodb\connection\server.js:472:18,     at MongoReply.parseBody (D:\6\node\fablab\node_modules\mongoose\node_modules\mongodb\lib\mongodb\responses\mongo_reply.js:68:5),     at null.<anonymous> (D:\6\node\fablab\node_modules\mongoose\node_modules\mongodb\lib\mongodb\connection\server.js:430:20),     at EventEmitter.emit (events.js:95:17),     at null.<anonymous> (D:\6\node\fablab\node_modules\mongoose\node_modules\mongodb\lib\mongodb\connection\connection_pool.js:191:13)]
